### ISSUE TITLE
unable to set `region` attribute for `aws` provider in a module

### ISSUE THREAD

**Issue by gosuri** [hashicorp/terraform#1186]
When in a module, `aws` provider does not read the `region` attribute with the below configuration:
I uploaded the source [here](https://github.com/gosuri/terraform-040-test).

Contents of `modules/dc/main.tf`:
```
provider "aws" {
  region = "us-west-1"
}
```
Contents on `main.tf`:
```
module "dc_ca1" {
  source = "modules/dc"
}
```

Running apply results in a error:
```sh
$ TF_LOG=1 terraform apply -input=false

2015/03/11 17:03:53 [INFO] Terraform version: 0.4.0 dev a24c21bd2c1d17e2a92b972dbfa56fc8260cd607
2015/03/11 17:03:53 Detected home directory from env var: /Users/gosuri
...
2015/03/11 17:03:53 [DEBUG] /Users/gosuri/projects/go/bin/terraform-provider-aws: plugin process exited
  * provider.aws: "region": required field is not set
```

See [complete log](https://raw.githubusercontent.com/gosuri/terraform-040-test/master/apply.log)

**Comment by jwestboston** [comment]
+1

**Comment by mitchellh** [comment]
This was fixed with #1282

**Comment by gosuri** [comment]
:+1:  Awesomness
